scipy/stats/tests/test_discrete_distns.py

```diff
@@ -29,6 +29,34 @@ def test_hypergeom_logpmf():
     assert_almost_equal(hypergeom_logpmf, bernoulli_logpmf, decimal=12)
 
 
+def test_nhypergeom_logpmf():
+    # test with hypergeom
+    N, K, r = 45, 13, 8
+    k = 6
+    NHG = nhypergeom.pmf(k, N, K, r)
+    HG = hypergeom.pmf(k, N, K, k+r-1) * (N - K - (r-1)) / (N - (k+r-1))
+    assert_allclose(HG, NHG, rtol=10)
+
+    # test pmf and cdf with arbitrary values.
+    N = 8
+    K = 3
+    r = 4
+    support = np.arange(K+1)
+    pmf = nhypergeom.pmf(support, N, K, r)
+    assert_allclose(pmf, [1/14, 3/14, 5/14, 5/14], rtol=1e-13)
+    cdf = nhypergeom.cdf(support, N, K, r)
+    assert_allclose(cdf, [1/14, 4/14, 9/14, 1.0], rtol=1e-13)
+
+    # test with r=0.
+    N = 10
+    K = 3
+    r = 0
+    with suppress_warnings() as sup:
```

Comment:
If you make the change suggested above to use `_lazywhere`, no warning should be generated, and this `suppress_warnings` context manager won't be necessary.